docs/dev_guide/ceph_s3_tests_guide.md

```diff
@@ -0,0 +1,145 @@
+Following the General Settings For Ceph S3 Tests.
+
+### 2) Increasing Debug Level (Noobaa-Operator)
+Before running a test, you can increase the debug level with noobaa CLI.
+```bash
+nb system set-debug-level <level>
+```
+A good level to start with is 2 or 3. the higher you go the more verbose and noisy the logs will become.
+
+### 3) Deploy The Tester Deployment 
+Use the `test_ceph_s3_deployment.yml` file to install the tester pod. This will install a pod that does nothing, but it allows us to go into the pod and run our test.
+```bash
+kubectl apply -f src/test/system_tests/ceph_s3_tests/test_ceph_s3_deployment.yml
+```
+
+### 4) Setup Test Config (Inside The Tester Pod)
+Once the tester pod is up, we can go into it and prepare the environment to run the tests.
+both `kubectl` and `oc` can be used:
+```bash
+# if you have kubectl 
+kubectl exec -it [noobaa-tester pod] -- bash
+# or with oc
+oc rsh [noobaa-tester pod] bash
+```
+
+In the tester pod, go to noobaa working directory:
+```bash
+cd /root/node_modules/noobaa-core/
+```
+
+Run the script that will create the necessary accounts in noobaa and update the Ceph S3 tests config file accordingly:
+```bash
+node ./src/test/system_tests/ceph_s3_tests/test_ceph_s3_config_setup.js
+```
+
+Note: If you want to ignore PythonDeprecationWarnings use (which will then ignore all Python warnings, so keep that in mind):
+```bash
+export PYTHONWARNINGS="ignore"
+```
+
+### 5) Run a Test (Inside The Tester Pod)
+To run a test, from noobaa working directory:
+```bash
+S3TEST_CONF=src/test/system_tests/ceph_s3_tests/test_ceph_s3_config.conf ./src/test/system_tests/ceph_s3_tests/s3-tests/virtualenv/bin/nosetests <test_name>
+```
+This should run the test on the noobaa deployment we've set up.
+
+### 6) Debug a Test (Inside The Tester Pod)
+You can view the test (python) by going to the test file and searching for the test function. e.g. if you are working on test `s3tests_boto3.functional.test_s3:test_set_bucket_tagging` then you should `vim ./src/test/system_tests/ceph_s3_tests/s3-tests/s3tests_boto3/functional/test_s3.py` and search for the function `test_set_bucket_tagging`.
+
+The best place to start investigating is noobaa endpoint logs. if you are running with debug level that is higher than 1, you should see messages of the S3 requests with the prefix `S3 REQUEST`. S3 replies will be with the prefix `HTTP REPLY`.
+
+### 7) Change a Test
+If there is a test that you suspect that it has a faulty and you would like to change the code, you will need to direct to your own forked repository. Since the file `./src/test/system_tests/ceph_s3_tests/s3-tests/s3tests_boto3/functional/test_s3.py` is a read-only file, follow the next steps:
+
+1) Fork and clone the repository [ceph/s3-test](https://github.com/ceph/s3-tests).
+2) Create a new branch from the hash number that was set in the file `./src/test/system_tests/ceph_s3_tests/test_ceph_s3_deploy.sh`.
+3) Change the code, commit and push to remote branch.
```

Comment:
@romayalon I didn't understand the comment, would you please add an additional explanation or suggestion?

Reply:
yes sure, I meant to say that I don't think I ever had to commit and push changes to the ceph tests repo in order to run this code and I think you can take the local repo instead of the remote one

Reply:
How do you edit this part by using a local repo?
https://github.com/noobaa/noobaa-core/blob/cd4afe250703a8a665c93072bf6c8ee9567a6320/src/test/system_tests/ceph_s3_tests/test_ceph_s3_deploy.sh#L16-L27

Reply:
I didn't do it for a while but CEPH_LINK can be the path to the local folder and instead of checking out a commit hash I think you can check out the used branch
it needs to be checked though

Reply:
@romayalon I added 2 options as discussed: Permanent change (the original instructions) and Temporary change (as Nadav did).